src/Core.UnitTests/EnvironmentSettingsTests.cs

```diff
@@ -103,5 +103,23 @@ public void SonarLintDaemonDownloadUrl_ReturnsExpectedValue(string envVarValue,
                 new EnvironmentSettings().SonarLintDaemonDownloadUrl().Should().Be(expected);
             }
         }
+
+        [TestMethod]
+        [DataRow(null, false)]
+        [DataRow("true", true)]
+        [DataRow("TRUE", true)]
+        [DataRow("false", false)]
+        [DataRow("FALSE", false)]
+        [DataRow("1", false)]
+        [DataRow("0", false)]
+        [DataRow("not a boolean", false)]
+        public void LogDebugMessages_CorrectlyMapped(string envVarValue, bool expected)
+        {
+            using (var scope = new EnvironmentVariableScope())
+            {
+                scope.SetVariable(EnvironmentSettings.LogDebugMessagesEnvVar, envVarValue);
+                new EnvironmentSettings().LogDebugMessages().Should().Be(expected);
+            }
+        }
```

Comment:
This test is copied from the `TreatBlockerSeverityAsError_CorrectlyMapped` test above.